WordPress/gutenberg
# Components: refactor `ResizableBox` to pass `exhaustive-deps`

## What?
Updates the `ResizeBox` component to satisfy the `exhaustive-deps` eslint rule

## Why?
Part of the effort in https://github.com/WordPress/gutenberg/pull/41166 to apply `exhuastive-deps` to the Components package

## How?
`debounceUnsetMoveXY` and `onResize` were missing dependencies of the the `useResizeLabel` hook. Adding `onResize` (a passed in prop) will technically cause the effect to fire every time the hook is called, but that was already happening due to the `width` and `height` deps (unless we wanted to memoize those).

Now that it was a dependency, `debounceUnsetMoveXY` needed its own `useCallback`. This would, in turn, have needed `unsetMoveXY` as a dependency, leading to wrapping _that_ method in yet another `useCallback`.

Instead, I moved `unsetMoveXY` into the single newly created `useCallback`. Now that we have only one function handling all of the `unsetMoveXY` it makes more sense to call that simply `unsetMoveXY` instead of `debounceUnsetMoveXY`.

The end result is a single `unsetMoveXY` method containing its own debounce logic, all wrapped in a single `useCallback` for the benefit of the `useEffect`'s dependency array.

The only two dependencies for the new `useCallback` are `fadeTimeout` and `isAxisControlled`. While we could technically memoize `isAxisControlled` to prevent it triggering the `useCallback` on every render, we'll still have `fadeTimeout` which is a prop that likely won't be memoized before it's passed in, so I didn't see a benefit to an extra `useMemo` call.

## Testing Instructions
1. From your local Gutenberg directory, run `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/resizable-box/`
2. Confirm that the linter returns no errors
3. Run Storybook locally, confirm the components stories and/or docs still work as expected
